diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c2b7b4..aee0434 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,4 +97,4 @@
 option(SD_USE_SYSTEM_GGML            "sd: use system-installed GGML library" OFF)
-#option(SD_BUILD_SERVER               "sd: build server example"                           ON)
+option(SD_BUILD_SERVER               "sd: build server example" ${SD_STANDALONE})
-
+
 set(CMAKE_C_STANDARD 11)
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index dc8150a..0e83372 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -4 +4,3 @@ add_subdirectory(cli)
-add_subdirectory(server)
+if(SD_BUILD_SERVER)
+    add_subdirectory(server)
+endif()
